rush.json

```diff
@@ -1,37 +1,255 @@
+     * if the underlying package was already approved, this would imply that the typings
+     * are also approved.
+     */
     "ignoredNpmScopes": [ "@types" ]
   },
 
+  /**
+   * If you use Git as your version control system, this section has some additional
+   * optional features you can use.
+   */
   "gitPolicy": {
-    // Require GitHub scrubbed e-mails
+    /**
+     * Work at a big company?  Tired of finding Git commits at work with unprofessional Git
+     * emails such as "beer-lover@my-college.edu"?  Rush can validate people's Git email address
+     * before they get started.
+     *
+     * Define a list of regular expressions describing allowable e-mail patterns for Git commits.
+     * They are case-insensitive anchored JavaScript RegExps.  Example: ".*@example\.com"
+     */
     "allowedEmailRegExps": [
       "[^@]+@users\\.noreply\\.github\\.com",
       "travis@example\\.org"
     ],
+
+    /**
+     * When Rush reports that the address is malformed, the notice can include an example
+     * of a recommended email.  Make sure it conforms to one of the allowedEmailRegExps
+     * expressions.
+     */
     "sampleEmail": "mrexample@users.noreply.github.com"
   },
 
+  "repository": {
+    /**
+     * This setting is sometimes needed when using "rush change" with multiple Git remotes.
+     * It specifies the remote url for the official Git repository.  If this URL is provided,
+     * "rush change" will use it to find the right remote to compare against.
+     */
+    // "url": "https://github.com/Microsoft/rush-example"
+  },
+
+  /**
+   * Event hooks are customized script actions that Rush executes when specific events occur
+   */
+  "eventHooks": {
+    /**
+     * The list of shell commands to run before the Rush installation starts
+     */
+    "preRushInstall": [],
+
+    /**
+     * The list of shell commands to run after the Rush installation finishes
+     */
+    "postRushInstall": [],
+
+    /**
+     * The list of shell commands to run before the Rush build command starts
+     */
+    "preRushBuild": [],
+
+    /**
+     * The list of shell commands to run after the Rush build command finishes
+     */
+    "postRushBuild": [],
+  },
+
+  /**
+   * Rush can collect anonymous telemetry about everyday developer activity such as
+   * success/failure of installs, builds, and other operations.  You can use this to identify
+   * problems with your toolchain or Rush itself.  THIS TELEMETRY IS NOT SHARED WITH MICROSOFT.
+   * It is written into JSON files in the common/temp folder.  It's up to you to write scripts
+   * that read these JSON files and do something with them.  These scripts are typically registered
+   * in the "eventHooks" section.
+   */
+  // "telemetryEnabled": false,
+
+  /**
+   * Allows creation of hotfix changes. This feature is experimental so it is disabled by default.
+   */
+  // "hotfixChangeEnabled": false,
```

Comment:
>hotfixChangeEnabled [](start = 6, length = 19)

what is the point of this? I haven't seen before.

Reply:
It's an old experiment that was never graduated to being a proper feature, but which now has widespread usage.

---
In reply to: [217227703](https://github.com/Microsoft/rush-example/pull/2#discussion_r217227703) [](ancestors = 217227703)

Reply:
We should do something about that, but it's out of scope for this PR.

---
In reply to: [217232300](https://github.com/Microsoft/rush-example/pull/2#discussion_r217232300) [](ancestors = 217232300,217227703)